### PR TITLE
XWIKI-15944: Add ability to pass test resources to Docker-based tests

### PR DESCRIPTION
**Pull request by surli**
* Fix places of the test resources: they should be mounted in browser
container, not in servlet container

### CODE REVIEW

**Comment by surli** [comment]
@vmassol could you review this one please?

File: xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/TestConfiguration.java [hunk 1]
```diff
@@ -459,4 +461,13 @@ public String getOutputDirectory()
         }
         return outputDirectory;
     }
+
+    /**
+     * @return the path of the directory containing the test resources on the container.
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/TestConfiguration.java]: I guess you mean "in the container", right?

Comment (vmassol) [reply xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/TestConfiguration.java]: wait, what container? :) This missing.

File: xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/XWikiDockerExtension.java [hunk 2]
```diff
@@ -84,6 +84,8 @@
 
     private static final String SUPERADMIN = "superadmin";
 
+    private static final String TEST_RESOURCES_HOST_PATH = "target/test-classes";
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/XWikiDockerExtension.java]: Some comment to explain what path this is would be nice.

File: xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/XWikiDockerExtension.java [hunk 3]
```diff
@@ -247,7 +249,8 @@ private BrowserWebDriverContainer startBrowser(TestConfiguration testConfigurati
             .withDesiredCapabilities(testConfiguration.getBrowser().getCapabilities())
             .withNetwork(Network.SHARED)
             .withNetworkAliases("vnchost")
-            .withRecordingMode(BrowserWebDriverContainer.VncRecordingMode.SKIP, null);
+            .withRecordingMode(BrowserWebDriverContainer.VncRecordingMode.SKIP, null)
+            .withFileSystemBind(TEST_RESOURCES_HOST_PATH, testConfiguration.getTestResourcesPath());
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/XWikiDockerExtension.java]: This is not going to be performant, but it depends on the volume of data. In any case I'd recommend to do what we do for the servlet container.

Comment (vmassol) [reply xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/XWikiDockerExtension.java]: In any case this is missing some documentation too. See the doc you removed below from the Servlet container for example ;)

File: xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/TestConfiguration.java (hunk 1 above)
Comment (vmassol) [reply xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/TestConfiguration.java]: I don't think this is the right place. You should put this in the Browser enum probably (similar to what we do for the other enums - See what you removed from the ServletEngine enum).

File: xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/AbstractContainerExecutor.java [hunk 4]
```diff
@@ -51,4 +52,25 @@ protected void start(GenericContainer container, TestConfiguration testConfigura
...
     }
+
+    /**
+     * Utility method to mount a directory from the host to the container with some optimization for different OS.
+     * @param container the container on which to mount the directory.
+     * @param sourceDirectory the path of the directory on the host to mount in the container.
+     * @param targetDirectory the target where to mount the directory in the container.
+     * @since 10.11RC1
+     */
+    protected static void mountFromHostToContainer(GenericContainer container, String sourceDirectory,
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/AbstractContainerExecutor.java]: I don't like static too much (causes all sort of issues, including for testing when it sets some states).

**Pull request merged**